src/gt4py/next/common.py

```diff
@@ -896,9 +900,20 @@ def ndarray(self) -> Never:
     def asnumpy(self) -> Never:
         raise NotImplementedError()
 
+    def as_scalar(self) -> core_defs.IntegralScalar:
+        if self.domain.ndim != 0:
+            raise ValueError(
+                "'as_scalar' is only valid on 0-dimensional 'Field's, got a {self.domain.ndim}-dimensional 'Field'."
+            )
+        assert self._cur_index is not None
+        return self._cur_index
+
     @functools.cached_property
     def domain(self) -> Domain:
-        return Domain(dims=(self.dimension,), ranges=(UnitRange.infinite(),))
+        if self._cur_index is None:
+            return Domain(dims=(self.dimension,), ranges=(UnitRange.infinite(),))
+        else:
+            return Domain()
```

Comment:
@egparedes let's discuss these changes...

Reply:
ok, let's discuss it. Maybe the 0d-index should be a special case of the Domain class, not of CartesianConnectivity...